### ISSUE TITLE
Make "Stay on Tab" selection configurable

### ISSUE THREAD

**Issue by olivergrahl** [zammad/zammad#641]
Currently, the button left to the "Update" button on a ticket view always defaults to "Stay on Tab".

This is rarely my first choice. 99% of the time, I want to close the tab, and either move on to the next ticket or get back to the overview. It would be super helpful if that button would remember what it was previously set to. Or define the default setting in the agent's preferences. 

An example scenario:
* On a new tab I'd prefer the button to default to "Next in Overview".
* On the next ticket I choose "Stay on Tab" to keep it open.
* I go back to Overview and open a second ticket. The button defaults to "Next in Overview".
* Now I've got two tabs open.
* I switch to my first tab. The button should still be "Stay on Tab" there, so that I can make additional updates without closing this tab.
* I switch to my second tab, and the button here is still the "Next in Overview". 

Basically, with this example, I'm saying that just syncing the buttons on all tabs would not be a favorable solution, better make the default customizable. Some may prefer "Close Tab" as the default as well. Alternatively, to avoid having another preference, the last selection could be stored and made the default. But thinking about it, a preference would be the better option.

Also, "Next in Overview" should close the tab when there is no additional ticket in the overview. At least, that's what I would have expected, but it seems to keep it open.

HTH.

**Comment by olivergrahl** [comment]
Thinking about it a little more, it might alternatively make sense to have an option like "Automatically close Tab when Ticket Status is set to Closed", which would disregard the "Stay on Tab" setting. In my workflow, when I close a ticket, I never want to keep it open (Heck, I _closed_ it!). It's a bit annoying having that extra click to close the tab, almost every time I reply to a ticket (and close it via a Macro).

**Comment by AND51** [comment]
> Alternatively, to avoid having another preference, the last selection could be stored and made the default. But thinking about it, a preference would be the better option.

I agree as currently the "stay on tab" or "close tab" does not work. It always jumps back to "stay on tab".

> In my workflow, when I close a ticket, I never want to keep it open

Me too. I press ctrl+alt+c to set the ticket to "closed" and then crtl+alt+w to close the tab. That's faster than clicking here and clicking there. Maybe that suits your needs?

**Comment by olivergrahl** [comment]
After using Zammad for quite some time now, I came to the conclusion that the "Stay on Tab" button is completely useless: Using the button, it takes **two** clicks to set this preference before updating the ticket, versus a **single** click to close the tab afterwards. While ideally, there sould be **zero** clicks to close the view after answering/closing a ticket.

I would suggest to either ...
- remove the button, 
- make its default customizable, 
- or make it default to its last selection.

**Comment by chrisblech** [comment]
+1

**Comment by raunokulla** [comment]
+1

**Comment by thorsteneckel** [comment]
Locking conversation. There is no new input. +1 and requesting comments create noise and distraction. We are working hard on the next upcoming/most requested features and projects of our support customers. If you want to state your interest in the feature and give valuable input please give the initial post a 👍 reaction and we are able to evaluate it. We will lock all distracting issues from now on. Thanks for your understanding.

**Comment by martini** [comment]
Hi @all

after some discussions with users it went out, that the best solution for this issue is to make it configurable based on macros. The  reason for that is, you do not want to have a default behaviour (e. g. always close the tab) because you usually only want to close the tab if the next ticket state is closed and not by just sending an update to the customer.

With next Zammad version you can define on macros what the next tab behaviour is.

In case, just try it and give feedback. Thanks in advance!

 -Martin
